src/Ocelot.Provider.Consul/PollConsul.cs

```diff
@@ -41,13 +51,22 @@ public void Dispose()
         _timer = null;
     }
 
-    public Task<List<Service>> Get()
+    public async Task<List<Service>> Get()
     {
-        return Task.FromResult(_services);
+        if (_cache.TryGetValue<List<Service>>(_serviceCacheKey, out var services))
+        {
+            return services;
+        }
+
+        await Poll();
+
+        return _cache.Get<List<Service>>(_serviceCacheKey);
     }
 
     private async Task Poll()
     {
-        _services = await _consulServiceDiscoveryProvider.Get();
```

Comment:
The "Poll" method mixes 2 things: getting services and caching. I think that respecting the principle of single responsibility, and putting caching in the main "Get" method, would make the code more readable.
Going further, is the "Poll" method really useful?

Reply:
Ray, is caching a good functionality at all in the provider?